lib/AST/LocalizationFormat.cpp

```diff
@@ -46,6 +46,10 @@ template <> struct ScalarEnumerationTraits<LocalDiagID> {
 #define DIAG(KIND, ID, Options, Text, Signature)                               \
   io.enumCase(value, #ID, LocalDiagID::ID);
 #include "swift/AST/DiagnosticsAll.def"
+    // Ignore diagnostic IDs that are available in the YAML file and not
+    // available in the `.def` file.
+    if (!io.outputting() && io.matchEnumFallback())
```

Comment:
What does `!io.outputting()` mean?

Reply:
Does it mean that caller is currently injecting YAML instead of forming one?

Reply:
If so then I guess it shouldn't matter right? Either way "missing" ids shouldn't be included...

Reply:
Yes, you're right. It's not needed